File: GameEngine_PC/GameEngineOpenGL3_2/src/main/java/gameEngine/models/AudioBuffer.java
package gameEngine.models;

/**
 * Audio buffer to get played
 */
public class AudioBuffer {

	/**
	 * Identifier of the buffer to play in openAL
	 */
	private int bufferId;

	/**
	 * @param bufferId
	 *            Identifier of the buffer to play in openAL
	 */
	public AudioBuffer(int bufferId) {
		super();
		this.bufferId = bufferId;
	}

	/**
	 * @return the bufferId Identifier of the buffer to play in openAL
	 */
	public int getBufferId() {
		return bufferId;
	}

	/**
	 * @param bufferId
	 *            Identifier of the buffer to play in openAL
	 */
	public void setBufferId(int bufferId) {
		this.bufferId = bufferId;
	}

}

File: GameEngine_IOS/GameEngine/AudioEngine/Models/AudioBuffer.swift
import Foundation

import OpenAL


/// Audio buffer to get played
open class AudioBuffer : NSObject {
    
    
    ///Identifier of the buffer to play in openAL
    fileprivate var bufferId : ALint;
    
    
    /// Initializer of the audio buffer
    ///
    /// - Parameter bufferId: Identifier of the buffer to play in openAL
    public init(_ bufferId : ALint) {
        self.bufferId = bufferId;
    }
    
    
    /// Get the identifier of the associated buffer
    ///
    /// - Returns: The Identifier of the buffer to play in openAL
    open func getBufferId() -> ALint {
        return bufferId;
    }
    
    
    /// Identifier of the buffer to play in openAL
    ///
    /// - Parameter bufferId: The identifier of the buffer
    open func setBufferId(_ bufferId : ALint) {
        self.bufferId = bufferId;
    }
    
}
